File: android/src/main/java/expo/modules/gaodemap/overlays/HeatMapViewModule.kt
package expo.modules.gaodemap.overlays

import expo.modules.kotlin.modules.Module
import expo.modules.kotlin.modules.ModuleDefinition

/**
 * HeatMap 视图 Module
 */
class HeatMapViewModule : Module() {
  override fun definition() = ModuleDefinition {
    Name("HeatMapView")

    View(HeatMapView::class) {
      Prop<List<Map<String, Any>>>("data") { view: HeatMapView, data ->
        view.setData(data)
      }
      
      Prop<Int>("radius") { view: HeatMapView, radius ->
        view.setRadius(radius)
      }
      
      Prop<Double>("opacity") { view: HeatMapView, opacity ->
        view.setOpacity(opacity)
      }
    }
  }
}

File: ios/overlays/HeatMapViewModule.swift
import ExpoModulesCore

public class HeatMapViewModule: Module {
    public func definition() -> ModuleDefinition {
        Name("HeatMapView")
        
        View(HeatMapView.self) {
            Prop("data") { (view: HeatMapView, data: [[String: Any]]) in
                view.setData(data)
            }
            
            Prop("radius") { (view: HeatMapView, radius: Int) in
                view.setRadius(radius)
            }
            
            Prop("opacity") { (view: HeatMapView, opacity: Double) in
                view.setOpacity(opacity)
            }
        }
    }
}